engine-tests/src/test/java/org/terasology/config/flexible/FlexibleConfigTest.java

```diff
@@ -0,0 +1,154 @@
+
+            Setting<Integer> retrievedSetting1 = config.get(id1);
+            Setting<Double> retrievedSetting2 = config.get(id2);
+
+            // We need the references to be equal
+            assertEquals(expectedSetting1, retrievedSetting1);
+            assertEquals(expectedSetting2, retrievedSetting2);
+        }
+    }
+
+    public static class Contains {
+        private FlexibleConfig config;
+
+        @Before
+        public void setUp() throws Exception {
+            config = new FlexibleConfigImpl();
+        }
+
+        @Test
+        public void testContains() throws Exception {
+            SimpleUri id = new SimpleUri("engine-tests:TestSetting");
+
+            config.add(new MockSetting<Integer>(id));
+
+            assertTrue(config.contains(id));
+        }
+
+        @Test
+        public void testNotContains() throws Exception {
+            SimpleUri id = new SimpleUri("engine-tests:TestSetting");
+
+            config.add(new MockSetting<Integer>(id));
+
+            assertFalse(config.contains(KEY_NON_EXISTENT));
+        }
+    }
+
+    public static class Add {
+        private FlexibleConfig config;
+
+        @Before
+        public void setUp() throws Exception {
+            config = new FlexibleConfigImpl();
+        }
+
+        @Test
+        public void testAdd() throws Exception {
+            SimpleUri id = new SimpleUri("engine-tests:TestSetting");
+
+            assertTrue(config.add(new MockSetting(id)));
+        }
+
+        @Test
+        public void testAddExisting() throws Exception {
+            SimpleUri id = new SimpleUri("engine-tests:TestSetting");
+
+            assertTrue(config.add(new MockSetting<Integer>(id)));
+
+            assertFalse(config.add(new MockSetting<Double>(id)));
+        }
+    }
+
+    public static class Remove {
+        private FlexibleConfig config;
+
+        @Before
+        public void setUp() throws Exception {
+            config = new FlexibleConfigImpl();
+        }
+
+        @Test
+        public void testRemove() throws Exception {
+            SimpleUri id1 = new SimpleUri("engine-tests:TestSetting1");
+            SimpleUri id2 = new SimpleUri("engine-tests:TestSetting2");
+
+            config.add(new MockSetting(id1));
+
+            config.add(new MockSetting(id2));
+
+            assertTrue(config.remove(id1));
+            assertTrue(config.remove(id2));
+        }
+
+        @Test
+        public void testNonexistentRemove() throws Exception {
+            SimpleUri id = new SimpleUri("engine-tests:TestSetting");
+
+            config.add(new MockSetting(id));
+
+            assertFalse(config.remove(KEY_NON_EXISTENT));
+        }
+
+        @Test
+        public void testSubscribedRemove() throws Exception {
+            SimpleUri id = new SimpleUri("engine-tests:TestSetting");
+
+            Setting setting = new MockSetting(id);
+
+            config.add(setting);
+
+            setting.subscribe(propertyChangeEvent -> {
+            });
```

Comment:
@emanuele3d I let the IDE/editor do the formatting for me because it has _never_ failed me, and because open source projects (like this one) usually define a set of formatting rules for their IDE of choice following the project's conventions which I don't want to deviate from.